packages/ckeditor5-bookmark/tests/bookmarkediting.js

```diff
@@ -81,6 +83,17 @@ describe( 'BookmarkEditing', () => {
 			);
 		} );
 
+		it( 'should properly downcast bookmark with and space text before', () => {
+			setModelData( model, '<paragraph>text <bookmark bookmarkId="foo"></bookmark></paragraph>' );
+
+			expect( editor.getData() ).to.equal(
+				'<p>' +
+					'text' +
```

Comment:
As we debugged it together, the issue relies inside `SetModelData` function, which strips the spaces while setting the data.

For now the workaround is instead of:

```js
setModelData( model, '<paragraph>text <bookmark bookmarkId="foo"></bookmark></paragraph>' );
```

is to put the selectrion marker after the space

```js
setModelData( model, '<paragraph>text []<bookmark bookmarkId="foo"></bookmark></paragraph>' );
```